web/text_layer_builder.js

```diff
@@ -40,41 +41,45 @@ import { renderTextLayer } from "pdfjs-lib";
  */
 class TextLayerBuilder {
   constructor({
-    textLayerDiv,
     eventBus,
     pageIndex,
     viewport,
     highlighter = null,
     accessibilityManager = null,
+    isOffscreenCanvasSupported = true,
   }) {
-    this.textLayerDiv = textLayerDiv;
     this.eventBus = eventBus;
     this.textContent = null;
     this.textContentItemsStr = [];
     this.textContentStream = null;
     this.renderingDone = false;
     this.pageNumber = pageIndex + 1;
-    this.viewport = viewport;
     this.textDivs = [];
+    this.textDivProperties = new WeakMap();
     this.textLayerRenderTask = null;
     this.highlighter = highlighter;
     this.accessibilityManager = accessibilityManager;
+    this.isOffscreenCanvasSupported = isOffscreenCanvasSupported;
+    this.scale = 0;
 
-    this.#bindMouse();
+    this.div = document.createElement("div");
+    this.div.className = "textLayer";
+    this.#setDimensions(viewport);
+    this.#update(viewport);
```

Comment:
Isn't there a risk at least the rotation, and potentially also the dimensions, could have changed between the constructor being invoked and the `render` method running?
Shouldn't we rather call `this.#update(viewport);` *unconditionally* in `render`-method?